vowpalwabbit/core/tests/automl_test.cc

```diff
@@ -103,7 +102,7 @@ get_automl_data(VW::workspace& all)
   all.l->get_enabled_reductions(e_r);
   if (std::find(e_r.begin(), e_r.end(), "automl") == e_r.end())
   {
-    BOOST_FAIL("automl not found in enabled reductions");
+    EXPECT_THROW("automl not found in enabled reductions", char*);
```

Comment:
```suggestion
    FAIL() << "automl not found in enabled reductions";
```

Reply:
`FAIL()` seems to only work for void functions, updated to `THROW`